cli/cli/helpers/github_auth_store/github_auth_store.go

```diff
@@ -0,0 +1,361 @@
+package github_auth_store
+
+import (
+	"errors"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/host_machine_directories"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"github.com/zalando/go-keyring"
+	"os"
+	"sync"
+)
+
+const (
+	//!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+	// DO NOT CHANGE THIS VALUE
+	// Changing this value could leak tokens in a users keyring/make Kurtosis unable to retrieve/remove them.
+	//!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+	kurtosisCliKeyringServiceName = "kurtosis-cli"
+
+	githubAuthFilesPerms = 0644
+)
+
+var (
+	// NOTE: This will be initialized exactly once (singleton pattern)
+	githubAuthStore GitHubAuthStore
+	once            sync.Once
+
+	NoTokenFound = errors.New("no token found for currently logged in user")
+)
+
+// GitHubAuthStore stores information about a GitHub user that has authorized Kurtosis CLI to perform git operations on their behalf
+// [username] is their GitHub username
+// [authToken] is a scoped token that authorizes Kurtosis CLI on behalf of [username
+type GitHubAuthStore interface {
```

Comment:
hopefully the interface comments + tests document the behavior of the code well. the only complexity is really determining where to store/retrieve auth token from